### DISCUSSION TITLE
长轮询有配置更新时，远程请求拉取配置失败的一致性问题

### DISCUSSION THREAD

**Discussion by oscarjm** [apolloconfig/apollo#4421] [Q&A]
长轮询发现客户端与服务端版本不一致时，在触发sync()前已经把本地的版本号（m_notifications）更新为远程的新版本号，如果sync在重试后依然失败，那是否意味着客户端短时间内没拿到新配置，且长轮询也无法发现本地配置版本落后，只能通过schedulePeriodicRefresh 定时刷新的机制来保证最终一致性。
疑问代码：

         final HttpResponse<List<ApolloConfigNotification>> response =
            m_httpClient.doGet(request, m_responseType);
        logger.debug("Long polling response: {}, url: {}", response.getStatusCode(), url);
        if (response.getStatusCode() == 200 && response.getBody() != null) {
          updateNotifications(response.getBody());
          updateRemoteNotifications(response.getBody());
          transaction.addData("Result", response.getBody().toString());
          notify(lastServiceDto, response.getBody());
        }

**Comment by nobodyiam** [comment]
I believe this is the designed behavior.
